sdk/src/client/api/block_builder/input_selection/requirement/sender.rs

```diff
@@ -42,6 +42,30 @@ impl InputSelection {
                     Err(e) => Err(e),
                 }
             }
+            // TODO https://github.com/iotaledger/iota-sdk/issues/1721
+            Address::Multi(multi_address) => {
+                let mut cumulative_weight = 0;
+
+                for weight_address in multi_address.addresses() {
+                    for input in self.selected_inputs.iter() {
+                        let required_address = input
+                            .output
+                            .required_and_unlocked_address(self.slot_index, input.output_id())?
+                            .0;
+
+                        if &required_address == weight_address.address() {
+                            cumulative_weight += weight_address.weight() as u16;
+                            break;
+                        }
+                    }
+                }
```

Comment:
could be done more efficiently

Reply:
I would probably also check the threshold in the loop so we can break if the threshold is surpassed already, unless we really need the exact amount of cumulative weight.

Reply:
Will do